packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart

```diff
@@ -916,35 +999,55 @@ class CpuProfileStore {
 
 class _CpuProfileTimelineTree {
   factory _CpuProfileTimelineTree.fromCpuSamples(
-    vm_service.CpuSamples cpuSamples,
-  ) {
-    final root = _CpuProfileTimelineTree._fromIndex(cpuSamples, kRootIndex);
+    vm_service.CpuSamples cpuSamples, {
+    bool buildCodeTree = false,
+  }) {
+    final root = _CpuProfileTimelineTree._fromIndex(
+      cpuSamples,
+      kRootIndex,
+      buildCodeTree,
+    );
     _CpuProfileTimelineTree current;
     // TODO(bkonyi): handle truncated?
-    for (final sample in cpuSamples.samples ?? []) {
+    for (final sample in cpuSamples.samples ?? <vm_service.CpuSample>[]) {
       current = root;
+      final stack = buildCodeTree ? sample.codeStack : sample.stack!;
```

Comment:
if buildCodeTree is true, do we need to still build the function tree as well? or is it an either / or situation depending on the value of buildCodeTree?

Reply:
This is an either or situation. We'll build two `_CpuProfileTimelineTree` instances in the process of building two `CpuProfileData` instances when VM developer mode is enabled. This seemed like the least intrusive way to share the logic for function and code profiles since it doesn't involve hacking at `CpuProfileData` to have it conditionally contain two profile types.